motor_control/throttle/button_control.ino

```diff
@@ -4,26 +4,62 @@ ButtonThrottle::ButtonThrottle(const int &pin, int timeToFull) : Throttle(pin, t
 {
   this->m_timeToFull = timeToFull;
   m_isPressed = false;
+  m_maxValue = 255;
+  m_speedOfCar = 0;
+  m_delaySpeed = 1;
 }
 
 int ButtonThrottle::mappingFunction(int time_pressed, m_map_type mapType)
 {
   if (mapType == LINEAR)
   {
-    return std::min(time_pressed / m_timeToFull, 1) * 100;
+    return std::min((time_pressed / m_timeToFull, 1.0) * m_maxValue;
   }
-  if (mapType == EXPONENTIAL)
+  if (mapType == POWER)
   {
+    if (time_pressed <= m_timeToFull)
+      {
+        return std::pow(time_pressed / m_timeToFull, 2) * m_maxValue; // This function uses m_timeToFull to create a dependency on what time we pick and creates an exponential to the power of the 2 with the ratio
+      }
+      else
+      {
+        return m_maxValue;
+      }
   }
-  if (mapType == LOGARITHMIC)
+  if (mapType == EXPONENTIAL)
   {
+     if (time_pressed < m_timeToFull)
+      {
+        double ratio = 0;
+        ratio = time_pressed / 1000;
+        if (ratio == 0)
+        {
+          return 0;
+        }
+        if (std::exp(ratio) <= m_maxValue)
+        {
+          return std::exp(ratio); //This function has no reliance on m_timeToFull, it simply follows an e exponential curve up til maximum value. Takes about 5.5 seconds to reach max speed(assuming 255 is the value we set)
+        }
+      }
+      else
+      {
+        return m_maxValue;
+      }
   }
 }
 
 /*
    * Reads the throttle input from analog signal,
    * Returns 0~100
    */
+                   
+int ButtonThrottle::mappingFunctionDeceleration(int val)
+{
+    int timeNotPressed = millis() - m_UnpressStart;
+    return val/(timeNotPressed*m_decelerationFactor);
+    
+}
+                    
 void ButtonThrottle::compute_motor_value()
 {
   if (get_raw_value() == 1)
```

Comment:
I think the logic in this conditional might need to be rethought. Specifically, the second if statement checking `!m_isPressed`. This if statement will always be true since nowhere in button_control.ino is `m_isPressed` set to true, so the value `timePressed` will always be equal to however long it takes the computer to go from line 69 to line 71 which I don't think is what was intended?